src/setup_node_env/node_version_validator.js

```diff
@@ -37,18 +37,31 @@ var rawRequiredVersion = (pkg && pkg.engines && pkg.engines.node) || null;
 var requiredVersion = rawRequiredVersion ? 'v' + rawRequiredVersion : rawRequiredVersion;
 var currentVersionMajorMinorPatch = currentVersion.match(/^v(\d+)\.(\d+)\.(\d+)/);
 var requiredVersionMajorMinorPatch = requiredVersion.match(/^v(\d+)\.(\d+)\.(\d+)/);
+var currentMinorPatch = currentVersionMajorMinorPatch[2].concat(
+  '.',
+  currentVersionMajorMinorPatch[3]
+);
+var requiredMinorPatch = requiredVersionMajorMinorPatch[2].concat(
+  '.',
+  requiredVersionMajorMinorPatch[3]
+);
 var isVersionValid =
   currentVersionMajorMinorPatch[1] === requiredVersionMajorMinorPatch[1] &&
-  currentVersionMajorMinorPatch[2] === requiredVersionMajorMinorPatch[2] &&
-  parseInt(currentVersionMajorMinorPatch[3], 10) >= parseInt(requiredVersionMajorMinorPatch[3], 10);
+  parseFloat(currentMinorPatch) >= parseFloat(requiredMinorPatch);
+var isMinorValid = currentVersionMajorMinorPatch[2] === requiredVersionMajorMinorPatch[2];
 
 // Validates current the NodeJS version compatibility when OpenSearch Dashboards starts.
 if (!isVersionValid) {
```

Comment:
If `requiredVersion` was `14.20.20` and `currentVersion` was `14.20.3`, `isVersionValid` would be true and `isMinorValid` would also be true; this should not be accepted. We cannot use `parseFloat`.

A better option would be:

```suggestion
var version = {
    current: {
        major: currentVersionMajorMinorPatch[1],
        minor: parseInt(currentVersionMajorMinorPatch[2], 10),
        patch: parseInt(currentVersionMajorMinorPatch[3], 10),
    },
    required: {
        major: requiredVersionMajorMinorPatch[1],
        minor: parseInt(requiredVersionMajorMinorPatch[2], 10),
        patch: parseInt(requiredVersionMajorMinorPatch[3], 10),
    },
}

var isVersionValid =
  version.current.major === version.required.major &&
  version.current.minor === version.required.minor &&
  version.current.patch >= version.required.patch;

var isMinorValid =
    version.current.major === version.required.major &&
    version.current.minor > version.required.minor;
```

and then the condition below would be
```js
!isVersionValid && !isMinorValid
```

Reply:
@AMoo-Miki  thank you for advice

Reply:
It's strange but I couldn't accept your suggestions.

Reply:
I see this error "Applying suggestions on deleted lines is currently not supported. "